### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by upendras-ltts**
Feature Login:
Created login form with input 'Username' and 'Password'
Added the basic validation for form inputs
Sending entered 'userName' by end user to the statsig console api
Added logout feature to test with various users.

and Minor changes

Find the attached screen.

login
logout

### CODE REVIEW

File: React-Todo-list-main/src/App.js [hunk 1]
```diff
@@ -19,25 +15,58 @@ function App() {
...
+  );
+
+  /**
+   * Setting the user and flag from the login details
+   * @param {*} userName
+   */
+  const setUserDetails = (userName) => {
+    let localUser = {
+      userID: userName,
+      email: "upendra.singh@ltts.com",
```
Comment (tore-statsig) [inline React-Todo-list-main/src/App.js:34]: make a random email as well - username@todoapp.com

File: React-Todo-list-main/src/App.js [hunk 2]
```diff
@@ -19,25 +15,58 @@ function App() {
...
-      user={user}
-    >
-      <div className="App">
-        <TodoWrapper />
-      </div>
-    </StatsigProvider>
+    <div className="App">
+      {!isLoggedIn ? (
+        <Login setUser={setUserDetails} />
+      ) : (
```
Comment (tore-statsig) [inline React-Todo-list-main/src/App.js:57]: If the user is not logged in, still render the `StatsigProvider` - it just wont have a logged in user.  But youve handled the user state just fine:

<div className="App">
  <StatsigProvider
          sdkKey={CLIENT_SDK_KEY}
          waitForInitialization={true}
          options={{
            environment: { tier: "staging" },
          }}
          user={user}
        >
{!isLoggedIn ? (
        <Login setUser={setUserDetails} />
      ) : <TodoWrapper onLogout = {onLogout} />
</StatsigProvider>
```

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 3]
```diff
@@ -46,12 +56,13 @@ export const TodoWrapper = () => {
 
   /**
    * Setting the experiment and feature value
-   * 
+   *
    */
   useEffect(() => {
-    console.log(`Experiment Config: ${JSON.stringify(config)}`);
-    setSortingOrder(config.value.sort_order);
+    console.log(`Experiment Config: ${JSON.stringify(experimentConfig)}`);
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:62]: remove the console log

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 4]
```diff
@@ -46,12 +56,13 @@ export const TodoWrapper = () => {
...
+    setSortingOrder(experimentConfig.value.sort_order);
     setFeatureValue(value);
+    setDynamicValue(dynamicConfig.value);
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:65]: Dont have a state for the sort order and the dynamic config value and use a useEffect - just call `useConfig` and use that value directly

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 5]
```diff
@@ -200,13 +209,44 @@ export const TodoWrapper = () => {
...
+            flexDirection: "row",
+            alignItems: "center",
+            backgroundColor: `${dynamicValue.backgroundColor}`,
+          }}
+        >
+          <p style={{ fontSize: "1rem", color: `${dynamicValue.textColor}` }}>
+            {dynamicValue.message}
+          </p>
+        </div>
+      )}
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:248]: Create a separate react component for this `<WarningBanner />`

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 6]
```diff
@@ -13,31 +14,40 @@ import {
...
   const { value, isLoading } = useGate(FEATURE_GATE_1);
   /**
    * To get the experiment config from statsig console
    */
-  const { config } = useExperiment(EXPERIMENT_SORTING);
+  const { config: experimentConfig } = useExperiment(EXPERIMENT_SORTING);
+
+  /**
+   * To get the dynamic config
+   */
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:38]: remove comments like this - the code is the source of truth

File: React-Todo-list-main/src/Components/TodoWrapper.js [hunk 7]
```diff
@@ -200,13 +209,44 @@ export const TodoWrapper = () => {
...
+            marginTop: ".5rem",
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+            backgroundColor: `${dynamicValue.backgroundColor}`,
+          }}
+        >
+          <p style={{ fontSize: "1rem", color: `${dynamicValue.textColor}` }}>
+            {dynamicValue.message}
+          </p>
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Components/TodoWrapper.js:246]: use the `DynamicConfig.get("backgroundColor", "#FFFFFF");` get methods like @lttsNilesh has in android

File: React-Todo-list-main/src/App.js (hunk 2 above)
Comment (upendras-ltts) [reply React-Todo-list-main/src/App.js:57]: In this are case are initializing the provider twice, once without user and once with user?

File: React-Todo-list-main/src/App.js (hunk 1 above)
Comment (upendras-ltts) [reply React-Todo-list-main/src/App.js:34]: Added this dummy email id

File: React-Todo-list-main/src/Components/TodoWrapper.js (hunk 6 above)
Comment (upendras-ltts) [reply React-Todo-list-main/src/Components/TodoWrapper.js:38]: Agree removing it

File: React-Todo-list-main/src/Components/TodoWrapper.js (hunk 5 above)
Comment (upendras-ltts) [reply React-Todo-list-main/src/Components/TodoWrapper.js:248]: created different component for warning banner.

File: React-Todo-list-main/src/Components/TodoWrapper.js (hunk 7 above)
Comment (upendras-ltts) [reply React-Todo-list-main/src/Components/TodoWrapper.js:246]: Adding white as default text color, and app background color as default background color.

File: React-Todo-list-main/src/Constant.js [hunk 8]
```diff
@@ -9,6 +9,9 @@ export const TODO_CREATED = "TODO_CREATED";
 export const TODO_DELETED = "TODO_DELETED";
 export const TODO_LAST_VIEWED = "TODO_LAST_VIEWED";
 export const EXPERIMENT_SORTING = "item_sorting";
-export const SS_CLIENT_SDK_KEY = "client-wWQtJlfJfEnLSIuP6QcnTATHyGVHUnJ2URTGLOVrwrl";
+export const CLIENT_SDK_KEY = "client-wWQtJlfJfEnLSIuP6QcnTATHyGVHUnJ2URTGLOVrwrl";
```
Comment (tore-statsig) [inline React-Todo-list-main/src/Constant.js:12]: remove this other key

**Review by tore-statsig — APPROVED** [review]

**Pull request merged**